### PR TITLE
Fhebool boolean block

### PR DESCRIPTION
**Pull request by tmontaigu**
- This makes FheBool use integer::BooleanBlock internally.
- It makes comparisons (eq, ne, le, etc) return a FheBool instead of
  FheUint/FheInt.
- It also moves the if_then_else and cmux methods to FheBool.
- Adds casting from FheBool to FheUint/FheInt (but not from
  FheUint/FheInt to FheBool as we expect users to do `a.ne(0)`
  as its matches Rust)

BREAKING CHANGE:
    - Comparisons now return FheBool
    - if_then_else/cmux are now methods of FheBool.

### CODE REVIEW

File: tfhe/src/high_level_api/booleans/mod.rs [hunk 1]
```diff
@@ -159,7 +195,7 @@ impl FheTryEncrypt<bool, CompressedPublicKey> for crate::FheBool {
     fn try_encrypt(value: bool, key: &CompressedPublicKey) -> Result<Self, Self::Error> {
         let key = &key.key;
         let ciphertext = key.key.encrypt(u64::from(value));
-        Ok(Self::new(ciphertext))
+        Ok(Self::new(BooleanBlock::new_unchecked(ciphertext)))
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/booleans/mod.rs:198]: internally those ops still return a shortint ciphertext ?

Comment (tmontaigu) [reply tfhe/src/high_level_api/booleans/mod.rs:198]: Hum, Yeah I haven't added encryt_bool for the different PublicKey types of integer. Should I add them in this PR ?

Comment (IceTDrinker) [reply tfhe/src/high_level_api/booleans/mod.rs:198]: is it a large change ? otherwise it's fine to put it in a separate PR I would say

Comment (tmontaigu) [reply tfhe/src/high_level_api/booleans/mod.rs]: I added the 2 missing functions

**Review by IceTDrinker — APPROVED** [review]
Thanks a lot !

**Pull request merged**